backend/src/packages/orders/libs/helpers/combine-filters.ts

```diff
@@ -0,0 +1,21 @@
+import { type SQL, and, eq } from 'drizzle-orm';
+import { type PgColumn } from 'drizzle-orm/pg-core';
+
+import { type DatabaseSchema } from '~/libs/packages/database/schema/schema.js';
+
+type Table = DatabaseSchema[keyof DatabaseSchema];
+
+const combineFilters = <TSchema extends Table>(
+  schema: TSchema,
+  searchBy: Record<PgColumn['name'], string | number | null>,
+): SQL | undefined => {
```

Comment:
why undefined can be returned?

Reply:
It's can be undefined beacause drizzle-orm `and()` operator potentially can return undefined type if conditions not passed